Adds map generators to build mode

AnturK opened the pull request:
![space station 13 2015-05-21 192700](https://cloud.githubusercontent.com/assets/4047233/7754768/68753116-ffef-11e4-83af-6372650b4cb4.png)
- Adds generate area mode to build mode.
- Adds few variations of asteroid template
- Adds syndicate template
- Some minor changes to build mode.

Added to make preparing areas for events less time consuming.

Not sure if admin tools count for freeze but anyway this is not finished yet so DNM please.

RemieRichards commented:
Wonderful use of the system, I approve :+1:

RemieRichards commented on code/modules/admin/verbs/buildmode.dm:
```diff
@@ -174,6 +179,12 @@
+			if(AREA_BUILDMODE)
+				var/list/gen_paths = typesof(/datum/mapGenerator) - /datum/mapGenerator
+
+				var/type = input(usr,"Select Generator Type","Type") in gen_paths
```
``` as null|anything in gen_paths```
and then a check for ```if(!type) return ``` so we can quickly cancel.

Dennok commented:
Wonderful

Stickymayhem commented:
Could you add a shuttle one as well that uses simulated shuttle turfs and walls that are properly iconed?

Even a simple square outline would save so much fucking time in shuttle events.

AnturK commented:
Sure, I'll be finishing this tommorow will do that then.

AnturK commented:
This is finished for now. I'll add custom/circular shape areas later.

AnturK committed:
Adds area generators to build mode
Adds two generators : Asteroid and Syndicate
Syndicate templates
Finish syndicate templates
Moves border and repressurize to helpers.dm
Repressurize generator
Shuttle generator
Buildmode now resets the corners when switching modes/generators

AnturK committed:
Updates paths

optimumtact commented:
@RemieRichards merge?
